rfcs/001-ecosystem-serfice-credit-module.md

```diff
@@ -0,0 +1,159 @@
+- Credit Class ID
+- Issuer
+- Metadata
+- List of:
+  - Receiving account
+  - Tradable units
+  - Retired units
+
+This operation will return a new credit batch ID.
+
+#### Metadata
+
+Credit issuance requires a number of pieces of metadata to be provided. When a credit batch is issued, the issuer can provide as an argument arbitrary metadata bytes. It is intended that this metadata field be used for the following types of information:
+- Project Identifier
+- Geography (in the form of a geo-polygon)
+- Dates (Start & End Dates that this batch represents)
+
+The above fields may be included on-chain (serialized as a JSON string or protobuf), or linked to as off-chain data, in which case the metadata value would be a URI for resolving the off-chain data.
+
+#### Overlap Behavior
+
+It is generally illegal to issue a credit batch of the same credit class for overlapping polygons and dates although the blockchain state machine will not explicitly enforce this as off-chain auditing and slashing is a more efficient and robust way to do this.
+
+### Send
+
+The send operation transfers some units of a credit batch from the current holder to another account. Retired/burned credits cannot be transferred.
+
+Sends can be performed only by the account owner, or an authorized representative (via some external module like message authorization).
+
+### Retire/Burn
+
+The retire or burn operation retires some units of a credit batch that the signer of the transaction holds, making them non-transferrable.
+
+Retiring can be done by either the credit owner, or the credit issuer.
+
+# Rationale
+
+Include an overview of what tradeoffs exist when taking this approach, what benefits come from it, and/or what alternatives were considered.
+
+## Fractional NFT Design
+
+The design described above can be understood as a Fractional NFT (Non-Fungible Token) design where a credit vintage represents a non-fungible asset uniquely identified by its ID and some corresponding metadata. Each credit vintage then consists of a fixed total quantity of credits that are represented as fungible tokens.
+
+If credits were represented as fully fungible tokens, then this would mean all credits of a given credit class would be completely interchangeable with each other regardless of the vintage they came from. This may indeed be the desired behavior in many cases, but representing a credit class in this way would remove the ability to be able to uniquely track credits from a given vintage, which is important for auditing purposes.
+
+The other alternative, to preserve full auditability of all credits, would require a traditional NFT design, where each individual atomic credit unit (e.g. 1 ton of carbon) were represented by a unique ID as a non-fungible asset. While this would provide the most auditability, it drastically increases the technical complexity, and introduces usability issues (e.g. each individual credit would need to be traded as an individual transaction).
```

Comment:
```suggestion
The other alternative, to preserve full auditability of all credits, would require a traditional NFT design, where each individual atomic credit unit (e.g. 1 ton of carbon) would be represented by a unique ID as a non-fungible asset. While this would provide the most auditability, it drastically increases the technical complexity, and introduces usability issues (e.g. each individual credit would need to be traded as an individual transaction).
```